### PR TITLE
Feature/252 paginator

### LINKED ISSUE #252: Paginator
*AC*
* Recreate existing anx-react paginator (http://docspot.devnxs.net/projects/anx-react/latest/#Paginator) using Lucid components.

### PR DESCRIPTION
**Pull request by sodiumjoe**
~~Still some work to be done (captured in checklist below), but~~ ready for UX review and ~~preliminary~~ code review: http://docspot.devnxs.net/projects/lucid/feature-252-paginator/#/components/Paginator

fixes #252 

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
    - [x] Paginator view
    - [x] Paginator reducers
    - [x] Paginator selectors
    - [x] ArrowIcon view
    - [x] SingleSelect props.isSelectionHighlighted
    - [x] TextField props.onBlur
- [x] PR has one of the `semver-` labels
- [x] ~~Two~~ One core team engineer approval~~s~~
- [x] One core team UX approval
- [x] add selector support to buildHybridComponent

### CODE REVIEW

**Comment by halseyhemingway** [comment]
screen shot 2016-06-01 at 6 35 48 pm @sodiumjoe  Is there any way to not get this little selection marquee around the icon?

**Comment by sodiumjoe** [comment]
@halseyhemingway hm, i'm not seeing that, what browser are you using?

**Comment by halseyhemingway** [comment]
Firefox (40.0.1)

**Comment by sodiumjoe** [comment]
thanks, will investigate

**Comment by halseyhemingway** [comment]
screenflick movie @sodiumjoe  did a bit more investigation on my end and it seems all buttons are affected, not just this one. I think I should probably just approve this PR and file a bug. Sound ok?

File: src/components/Paginator/Paginator.jsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	object,
+	shape,
+	string,
+} = React.PropTypes;
+
+const { Option } = SingleSelect;
+
+/**
+ *
+ * {"categories": ["navigation"], "madeFrom": ["TextField", "Button", "SingleSelect"]}
```
Comment (jondlm) [inline src/components/Paginator/Paginator.jsx]: Missing `ArrowIcon`

Comment (sodiumjoe) [inline src/components/Paginator/Paginator.jsx]: i had fixed this, just forgot to push

File: src/components/Paginator/Paginator.selectors.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+function totalPagesSelector({
+	pageSizeOptions,
+	selectedPageSizeIndex,
+	totalCount,
+	totalPages,
+}) {
+	const pageSize = pageSizeOptions[selectedPageSizeIndex];
+	return _.isNumber(totalPages)
+		? totalPages
+		: totalCount <= pageSize
```
Comment (jondlm) [inline src/components/Paginator/Paginator.selectors.js]: For a second I thought this was a reverse fat arrow and I thought you were using some new feature of es6. I was gearing up for a google dive. Then I realized it's just a plain old less than or equal to. 👊

File: src/components/SingleSelect/SingleSelect.jsx [hunk 3]
```diff
@@ -174,7 +180,8 @@ const SingleSelect = createClass({
 			>
 				<DropMenu.Control>
 					<div className={cx('&-Control', {
-						'&-Control-is-selected': !isDisabled && isItemSelected,
+						'&-Control-is-highlighted': (!isDisabled && isItemSelected && isSelectionHighlighted) || (isExpanded && isSelectionHighlighted),
```
Comment (jondlm) [inline src/components/SingleSelect/SingleSelect.jsx]: If these two are exactly the same, why do we need both?

File: src/components/TextField/TextField.jsx [hunk 4]
```diff
@@ -64,6 +64,14 @@ const TextField = createClass({
 		 */
 		onChange: func,
 
+
+		/**
+		 * Fires an on the `input`'s onBlur.
+		 *
+		 * Signature: `(currentValue, { event, props }) => {}`
+		 */
+		onBlur: func,
```
Comment (jondlm) [inline src/components/TextField/TextField.jsx]: Noice

File: src/components/SingleSelect/SingleSelect.jsx (hunk 3 above)
Comment (sodiumjoe) [inline src/components/SingleSelect/SingleSelect.jsx]: renamed it from `-is-selected` to `-is-highlighted` but kept the former for backward compatibility. Have an issue with 2.0.0 milestone to deprecate it: https://github.com/appnexus/lucid/issues/289

**Pull request merged**